### PR TITLE
Updates to documentation

### PR DESCRIPTION
**Pull request by kian99**
#Description

- Remove mentions of Candid
- Remove old defunct docs

### CODE REVIEW

File: how-to/deploy_jimm_k8s.rst
Comment (kian99) [inline how-to/deploy_jimm_k8s.rst:1]: Not sure if this file should be removed. Let me know what you think.

File: explanation/jaas_security_scope.rst [hunk 1]
```diff
@@ -18,9 +18,9 @@ additionally acting as the IdP (Identity Provider) for said tokens.
 Access control and authentication
 ---------------------------------
 JAAS provides an abstraction layer of access control on top of Juju. JAAS does this by backing its users
-with an IdP (Identity Provider), this particular IdP is called Candid which handles the authentication 
-for JIMM (Juju Intelligent Model Manager, the back-end to JAAS). Furthermore, JAAS is  capable of many 
-authentication and authorisation types, ranging from users to agents. For authorisation, JAAS provides
+with an IdP (Identity Provider), many identity providers exist e.g. cloud providers like Google and Microsft can be used
```
Comment (babakks) [inline explanation/jaas_security_scope.rst:21]: Maybe rephrase it like this:

> JAAS does this by backing its users with an IdP (Identity Provider). Many identity providers can be used for this purpose (e.g. Google or Microsoft).

File: explanation/jaas_security_scope.rst [hunk 2]
```diff
@@ -18,9 +18,9 @@ additionally acting as the IdP (Identity Provider) for said tokens.
...
+as an IdP. We recommend the `Canonical identity platform <https://charmhub.io/topics/canonical-identity-platform>`__
```
Comment (babakks) [inline explanation/jaas_security_scope.rst:22]: Since we might use this link in a few other places, it's best to define it in the `links.txt` and reuse it everywhere.

File: explanation/jaas_security_scope.rst [hunk 3]
```diff
@@ -18,9 +18,9 @@ additionally acting as the IdP (Identity Provider) for said tokens.
...
+as the preferred IdP for JAAS. The IdP will handle user authentication on behalf of JAAS. For authorisation, JAAS provides
```
Comment (babakks) [inline explanation/jaas_security_scope.rst:23]: s/handle/handles

File: explanation/jaas_security_scope.rst [hunk 4]
```diff
@@ -18,9 +18,9 @@ additionally acting as the IdP (Identity Provider) for said tokens.
...
 this by means of tags and ReBAC (Relation Based Access Control).
```
Comment (babakks) [inline explanation/jaas_security_scope.rst:24]: s/Relation Based/Relation-Based

File: how-to/use_terraform.rst [hunk 5]
```diff
@@ -13,11 +13,11 @@ For this how-to you will need the following:
...
 - Client credentials (``client_id`` and ``client_secret``) generated by the above identity provider.
-- A deployed JIMM configured to trust the identity provider. For instructions on 
-  how to deploy JIMM read :doc:`deploy_jimm`.
-- A Juju 3.5 controller added to JIMM that can be used to control your chosen cloud. For instructions
+- A deployed JAAS configured to trust the identity provider. For instructions on 
+  how to deploy JIMM read :doc:`../tutorial/deploy_jaas_microk8s`.
+- A Juju controller added to JIMM that can be used to control your chosen cloud. For instructions
   on how to add one read :doc:`add_controller`.
-- A Juju 3.5 client.
+- A Juju client.
```
Comment (babakks) [inline how-to/use_terraform.rst:20]: Why not mentioning Juju 3.5? Actually, this should be 3.5.1 or higher.

File: how-to/use_terraform.rst [hunk 6]
```diff
@@ -46,7 +46,7 @@ Juju Terraform provider
 To authenticate with JIMM the provider section in your Terraform plan needs to include 
 the ``client_id`` and ``client_secret`` generated by your identity provider. Please note that 
 you need to use a version of the `Juju Terraform provider <https://registry.terraform.io/providers/juju/juju/latest/docs>`_
-higher than ``0.12.0``.
+higher or equal to ``0.12.0``.
```
Comment (babakks) [inline how-to/use_terraform.rst:49]: Definitely not 0.12.0, because it doesn't work. I asked about the next version from the Juju Terraform Provider team and the next version will be `0.13.0`.

**Review by babakks — APPROVED** [review]
Just a few comments.

File: how-to/use_terraform.rst (hunk 5 above)
Comment (kian99) [reply how-to/use_terraform.rst:20]: Because then it sounds like you can't use Juju 3.6. We explain the minimum versions required to use JAAS in a separate doc.

File: explanation/jaas_security_scope.rst (hunk 3 above)
Comment (kian99) [reply explanation/jaas_security_scope.rst:23]: I don't think that's correct here

**Review by ale8k — APPROVED** [review]
lgtm
